survey/decorators.py

```diff
@@ -19,8 +20,12 @@ def survey_check(self, request, *args, **kwargs):
         if not survey.is_published:
             raise Http404
         if survey.expire_date < date.today():
+            msg = "Survey is not published anymore. It was published until: {} ".format(survey.expire_date)
+            logging.warning(msg)
             raise Http404
         if survey.publish_date > date.today():
+            msg = "Survey is not yet published. It is due: {} ".format(survey.publish_date)
+            logging.warning(msg)
```

Comment:
Use lazy formatting so the formatting is not done unless we're in warning level of logging.
```suggestion
            msg = "Survey is not yet published. It is due: %s."
            logging.warning(msg, survey.publish_date)
```